### DISCUSSION TITLE
CMYK, CIELAB

### DISCUSSION THREAD

**Discussion by blue-j** [saucecontrol/Compact-ICC-Profiles#12] [Ideas]
What exactly is the state of CMYK profile support in browsers?  In my tests they always lack contrast. Is there a way to improve this or workaround?

CIELAB support is aspirational of course... no normal web formats even support it. Is there a new JPEG flavor or other format (AVIF?) that supports it?

Big thanks! 
J

**Comment by saucecontrol** [comment]
I'm not aware of any issues converting CMYK in Chromium or Firefox.  AFAIK, Chromium uses Skia for its CMYK conversion, so if you have examples where it fails to convert correctly to the display profile, it would be worth opening a bug with them.

I'm not sure what the use case would be for CIELAB on the web.  Web pages are viewed on RGB displays, so storing images in a format not easily convertible to RGB is wasteful, and CIELAB conversion is quite expensive.  Moreover, CIELAB really needs more than 8 bits per channel of precision to avoid banding, so there would be a storage penalty in addition to the processing penalty.

Interestingly, JPEG actually does allow any type of channel data to be encoded, so you could put CIELAB in it if any decoding app supported it.  The existing CMYK and RGB (as opposed to Y'CbCr) JPEG formats were initially abuses of the format by Adobe that others supported out of necessity for compatibility.  I don't think AVIF allows arbitrary channel definitions, but JPEG XL does, so that could work too -- if there were a compelling use case for it.

**Reply by blue-j** [reply]
Thanks so much for sharing your work and knowledge. I am very grateful.  : )

In all my tests CMYK rendering is broken in Safari and Chrome both.  Always.  It seems to be entirely tonal.  I have already filed a bug report, to no avail.  I'll attach one of my sample images along with a screenshot of Photoshop 2023's preview vs Chrome 107.0.5304.110.  Chrome is on the left; PS on the right.

![CMYK_sample_image](https://user-images.githubusercontent.com/3990142/202014808-c98e96f1-90a9-491f-819a-61a14c1ee5f1.jpg)
<img width="1638" alt="Screen Shot 2022-11-15 at 12 03 02 PM" src="https://user-images.githubusercontent.com/3990142/202014868-f240603e-5c21-4ece-bf0e-cbf08d068936.png">

I have tested many other images, with the same result.

The reason for my asking about CIELAB support is because I work on a web application that is very robust and has to handle all color models.  The line between desktop and browser has become very dotted.  It's not a pressing concern though, and your points are well made.

- J

**Reply by saucecontrol** [reply]
Looks to me like the conversion is done with different rendering intents.  Chrome seems to be using Relative Colorimetric intent and Photoshop seems to be using Perceptual intent.  You can choose which you want in PS when converting explicitly, but it makes sense Chrome would pick for you.  Not that either is necessarily right or wrong... the two colorspaces have different gamuts, so sacrifices have to be made somewhere.  I suppose that's one of the dangers of using CMYK on the web.

Funny side note, just look at what MS Outlook did to your original image in the notification email for your comment.
![image](https://user-images.githubusercontent.com/7929102/202021929-5b239863-3ec1-4552-8204-1d8ce551b1b3.png)

**Comment by blue-j** [comment]
It doesn't appear to be rendering intent.  I welcome you to dig deeper and
have a look!  I reserve the right to be wrong of course...

- J

**Reply by saucecontrol** [reply]
Yeah, I think you're right.  Did some playing with LittleCMS (which AFAICT Chrome is still using for CMYK) last week and noticed it doesn't produce the same black levels as Photoshop, even with the same rendering intent given explicitly.  I wrote my own basic CMYK->RGB implementation, and it's within rounding distance of Photoshop, so something is not right there.

Firefox doesn't even use the embedded profile for conversion, so its output is complete garbage.

**Comment by blue-j** [comment]
We can ask Marti about it?  Might be a BPC thing?  Maybe not, considering
your code didn't need it...?  Safari also fails, though not as badly...
which is surprising.  I am vexed.  It may be in the transform to monitor
RGB?  I'll ask around more.

- J
